### ISSUE TITLE
0.12upgrade remove comments and empty lines

### ISSUE THREAD

**Issue by Porkepix** [hashicorp/terraform#24422]
### Terraform Version
```
Terraform v0.12.24
```

Git diff outputs:
```diff
 resource "aws_route53_record" "txt" {
-  zone_id = "${aws_route53_zone.main.zone_id}"
+  zone_id = aws_route53_zone.main.zone_id
   name    = ""
   type    = "TXT"
   ttl     = "86400"

   records = [
     "v=spf1 include:aspmx.googlemail.com mx -all",
-
-    # Comment
     "google-site-verification=foobar",
-
-    # Comment
     "google-site-verification=foobar",
-
-    # Comment
     "google-site-verification=foobar",
   ]
 }
```

As seen above , the `0.12upgrade` command is removing both comments, and empty lines that were here to improve readability so that there are some "logical blocks".

I have other examples for removed empty lines. Comments should be preserved, and maybe also some of those lines. All of these cases didn't had any issue with `terraform fmt` on version `0.11`.

**Comment by jbardin** [comment]
Hi @Porkepix,

Thanks for reporting this. The `0.12upgrade` tool is now frozen to ensure that existing configurations are always upgraded consistently. We are focusing work on 0.13, and can take any improvement into consideration for future tooling. 

Thanks!

**Comment by Porkepix** [comment]
@jbardin Does that mean that there will be some `0.13upgrade` command? Or that it's unneeded and that people on `0.11` and earlier will still need to rely on `0.12upgrade` even to go on `0.13` and above? In the second case, I think it's still worth to consider fixing the present command.
In the second case, would this issue deserve consideration for hypothetical `0.13upgrade` command?

**Comment by jbardin** [comment]
There is likely to be a `0.13upgrade` tool to help upgrade to 0.13 features, similar to how the `0.12upgrade` tool operated. The `0.11` upgrade process will still require first upgrading to the latest `0.12` release in order to also make any local configuration and state stages necessary.

Upgrading a 0.11 configuration is meant to be a one-time process, often requiring manual intervention due to many ambiguities that cannot be resolved automatically. Formatting and comment issues exist because the 0.11 parse doesn't not provide enough information to relocate those properly, and some manual cleanup is expected there as well.

The `0.12` branch is no longer under active development, and only critical releases will be provided moving forward. The output also needs to remain consistent for organizations that crafted config to work in `0.11` and `0.12` through the upgrade process using the `0.12upgrade` tool in their automation, in which case the tool cannot risk any breaking changes to those workflows.

**Comment by Porkepix** [comment]
> Upgrading a 0.11 configuration is meant to be a one-time process, often requiring manual intervention due to many ambiguities that cannot be resolved automatically. Formatting and comment issues exist because the 0.11 parse doesn't not provide enough information to relocate those properly, and some manual cleanup is expected there as well.

Problem is that here it's beyond cleanup and solving ambiguities flagged by `TF-UPGRADE-TODO:` (which can easily and quickly be found) : user have to manually look at every single `git diff` to manually rewrite comments and if he want the lines for readability of files, one by ones, side by side with the diff and the file.

At the very least if it's impossible to fix this for `0.12`, I think it's worth considering it for `0.13` (even if for my part I think this is the kind of issue that would anyway always deserve a fix).

**Comment by Porkepix** [comment]
@jbardin Don't you think so about at the very least considering `0.13`?
